api/src/main/java/org/openmrs/module/fhir/api/comparator/AbstractPropertyComparator.java

```diff
@@ -0,0 +1,54 @@
+package org.openmrs.module.fhir.api.comparator;
+
+import org.hl7.fhir.dstu3.model.AllergyIntolerance;
+import org.hl7.fhir.dstu3.model.Base;
+
+import java.util.List;
+
+public abstract class AbstractPropertyComparator<B extends Base> {
+
+	protected abstract List<Boolean> compareProps(B b1, B b2);
```

Comment:
Maybe it will be better if this list will return one element (true if all props are equals). What do you think about that?